matchbox_socket/src/webrtc_socket/socket.rs

```diff
@@ -345,54 +388,76 @@ pub struct MessageLoopChannels {
     pub messages_from_peers_tx: Vec<futures_channel::mpsc::UnboundedSender<(PeerId, Packet)>>,
 }
 
+pub struct SocketChannels {
+    pub messages_from_peers_rx: Vec<UnboundedReceiver<(String, Box<[u8]>)>>,
+    pub new_connected_peers_rx: UnboundedReceiver<String>,
+    pub disconnected_peers_rx: UnboundedReceiver<String>,
+    pub peer_messages_out_tx: Vec<UnboundedSender<(String, Box<[u8]>)>>,
+}
+
+async fn run_signalling(
+    config: WebRtcSocketConfig,
+) -> Result<(MessageLoopChannels, SocketChannels), Error> {
+    let mut attempt = 0;
+    'signalling: loop {
+        info!("Signalling attempt #{attempt}");
+        let (requests_sender, requests_receiver) =
+            futures_channel::mpsc::unbounded::<PeerRequest>();
+        let (events_sender, events_receiver) = futures_channel::mpsc::unbounded::<PeerEvent>();
+        info!("a");
+        if let Err(e) =
+            signalling_loop(config.room_url.clone(), requests_receiver, events_sender).await
```

Comment:
This currently hangs infinitely because the messaging loop isn't running yet. I'm trying to figure out how to ensure the messaging loop can run concurrently during the signalling process.